stacks-signer/src/client/stackerdb.rs

```diff
@@ -80,7 +81,7 @@ impl StackerDB {
             if let Some(reason) = chunk_ack.reason {
                 // TODO: fix this jankiness. Update stackerdb to use an error code mapping instead of just a string
                 // See: https://github.com/stacks-network/stacks-blockchain/issues/3917
-                if reason == "Data for this slot and version already exist" {
+                if reason.contains("Data for this slot and version already exist") {
```

Comment:
Not necessary in this PR, but it would be good to have something like `StackerDBErrorCodes::from(reason)` or `from_json`. 

Until then, there is a `code` included now, so we can change this to `if reason.code == 0`

Reply:
Yes I tried to quickly add this. not as straightforward as I thought it would be because it actually does not return the struct....